### ISSUE TITLE
Can't export group messages before it was updated into a supergroup

### ISSUE THREAD

**Issue by ArmeF97** [telegramdesktop/tdesktop#5139]
### Steps to reproduce
1.  Open a supergroup that has got some messages before becoming a supergroup
2.  Export all messages

### Expected behaviour
All messages are exported

### Actual behaviour
Only messages after becoming a supergroup are exported

### Configuration
**Operating system:** Windows 10 v1803 (build 17134.254)

**Version of Telegram Desktop:** 1.3.14

**Used theme**: Default

**Comment by john-preston** [comment]
@ArmeF97 Yeah, old group is exported only when using full data data export.

**Comment by ArmeF97** [comment]
So it's a bug or a feauture? 😕 
Why can't we also export old messages from a single chat?

**Comment by john-preston** [comment]
@ArmeF97 It is an absence of a feature. It’s not trivial :( and was not implemented.

**Comment by ArmeF97** [comment]
I just tried the full export and it didn't export the messages before the upgrade 😢 
@john-preston

**Comment by john-preston** [comment]
@ArmeF97 It did. Just not in a united history. The old chat is a different chat from API point of view, so it was exported as a separate one.

**Comment by ArmeF97** [comment]
I'll search better and post here the result, thanks

**Comment by ArmeF97** [comment]
Yeah, @john-preston , it worked as you said.
I converted this issue in a feature request

**Comment by CoeusCC** [comment]
I can confirm the approach listed above works. What I didn't understand previously—and what I would assume is vexing to others—is that _you can't export the pre-supergroup messages in the one channel by itself in the channel settings_. You need to go to your **personal user settings** (the left-hand hamburger next to the search bar in the desktop app), then **Settings**, then **Advanced**, then **Export Telegram Data**.

I found the following link useful for making that connection, though it might help to make that more clear or explicit: https://telegram.org/blog/export-and-more

**Comment by CharisAsgard** [comment]
Chat History export do not export messages before conversion to supergroup

**Comment by icy** [comment]
I'm still interested in the feature and I have no idea how to export the past history of my supergroups.

**Comment by millerlogic** [comment]
All I want to do is export a group so I can delete it.
This 4 member group didn't even choose to "upgrade" and we still can't fully export.

**Comment by hbh7** [comment]
I don't think this was ever fixed (I haven't tried it in a long time anyways) so unless someone can verify it has been fixed, this should definitely stay open.

**Comment by john-preston** [comment]
@hbh7 It should be fixed. Unless it was broken after I've implemented the merging of the histories (legacy group + supergroup) in export.

**Comment by ice1000** [comment]
What about 'delete all from this user' for the messages before supergroup upgrade?

**Comment by icy** [comment]
> @hbh7 It should be fixed. Unless it was broken after I've implemented the merging of the histories (legacy group + supergroup) in export.

I have a family group dated before Feb 2016. Today I've found that the group only contains history from Feb 2016 and later days. It's a bit said to completely lost all past images and stories :( 

I have tried the "export function", but that doesn't help.  Is there any option that I can see the old messages?

**Comment by icy** [comment]
@hbh7 Can you please tell me how to "implement the feature"? I can spend time on that, if that's possible. Many thanks

**Comment by jentfoo** [comment]
After a supergroup conversion without warning, the lack of this feature is a severely impacting issue.  I really wish our channels history could be maintained in an export.  I am really disappointed this is considered a feature request and not a bug, beyond the significant user impact there is no graceful dialog or other indication that this is simply not supported right now.

I will be canceling my Telegram Plus over this issue specifically
